### DISCUSSION TITLE
Sync performance

### DISCUSSION THREAD

**Discussion by silviot** [librocco/librocco#354] [General]
I'm opening this discussion to have a space where we can go over ideas to improve the pouchdb sync.
The goal should be:
* Initial sync should be as fast as possible
* It should not load the CPU like it does now

I noticed that as the initial sync progresses, its performance degrades. Reloading the page helps in making it faster. Knowing this we could probably improve the current state by restarting the sync every 500 synced documents.

[Screencast from 06-09-23 11:18:12.webm](https://github.com/librocco/librocco/assets/396425/a02f8619-4fdd-4e5e-8373-48463624f3bb)

**Comment by ikusteu** [comment]
> I noticed that as the initial sync progresses, its performance degrades. Reloading the page helps in making it faster. Knowing this we could probably improve the current state by restarting the sync every 500 synced documents.

This could also mean that it clutters the tab memory with sync meta-rubbish...

**Comment by ikusteu** [comment]
One idea that might work, but might be a complete miss, is to use a service worker and move the sync off the main thread...(if that's even possible)
